### PR TITLE
Golang program for implementation of Merge Sort

### PR DESCRIPTION
**Pull request by vikumkbv**
**Hi!**  
Thanks for considering contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating Pull Requests
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   ✔ I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   ✔ I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   ✔ My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   ✔ This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   ✔ This pull request shall only be merged once the build passes.

-   ✔ I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-msrt/golang/mergesort.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Merge Sort in Golang
```
Comment (RavinduSachintha) [inline algorithms/ar-msrt/golang/mergesort.go:1]: Codacy found an issue: [package comment should be of the form "Package mergesort ..."](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278215)

File: algorithms/ar-msrt/golang/mergesort.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+// Merge Sort in Golang
+package mergesort
```
Comment (Danushka96) [inline algorithms/ar-msrt/golang/mergesort.go:2]: @vikumkbv add an empty line before the package

File: algorithms/ar-msrt/golang/mergesort.go (hunk 1 above)
Comment (RavinduSachintha) [inline algorithms/ar-msrt/golang/mergesort.go:1]: Codacy found an issue: [package comment should be of the form "Package mergesort ..."](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6278215)

**Comment by RavinduSachintha** [comment]
@vikumkbv we cannot accept this PR because,
- It is not related to any issue that you have assigned.
If you wish to contribute please follow the guidelines. Thank you.

**Pull request closed without merging**